envoy/network/socket.h

```diff
@@ -85,6 +85,12 @@ class ConnectionInfoProvider {
    **/
   virtual absl::optional<uint64_t> connectionID() const PURE;
 
+  /**
+   * @return the name of the network interface used by the downstream connection, or unset if not
```

Comment:
Looks like this interface is only used by the client connection. 
So i am not clear is that right to say `the name of the network interface used by the **downstream connection**`?

Reply:
I think this interface (ConnectionInfoProvider) is used for both, so maybe the language should be less specific + a TODO for setting this on server connections?

That way if someone wants to read the interface of the downstream server connection they'll run into the TODO and understand why it doesn't work

We seem to use the same language on `connectionID` but it does feel a bit confusing. The whole situation of there being a single StreamInfo object used for both upstream and downstream connections is a bit confusing, so maybe we should do some cleanup there separately.

Reply:
yep, we need some minor pr to clean these comments to make it more clear.